### ISSUE TITLE
Workflow engine does not allow for user hidden tags

### ISSUE THREAD

**Issue by mickenordin** [nextcloud/server#47636]
https://github.com/nextcloud/server/blob/d9ae08ba603c59d473de5be12ffaa6ea63667e9c/apps/workflowengine/lib/Check/FileSystemTags.php#L87

If an admin creates a tag that is not uservisible and shares that with a non-admin user, this part of the code throws an error. 

Why shouldnt it be allowed to have tags that is not visible to the user on a file or folder?

Maybe we can just remove that check?

**Comment by susnux** [comment]
cc @blizzz

**Comment by blizzz** [comment]
hidden tags should not leak or be guessable, that is the main behind the logic. But you can argue, of course, when the flow was set up by an admin, it should be allowed.

**Comment by mickenordin** [comment]
Yes, that is my thinking too. I can see that being useful, and not allowing it can trigger ui crashes, e.g. on the flow page for a user e.g. with automated tagging with a hidden tag.
